File: JavaScript/sharedSubstring.js
/*
    Returns true on finding a shared substring of length seekLength, else returns false.
*/

const checkForSharedSubstring = (primaryString, comparedString, seekLength) => {
    if (comparedString.length < seekLength) return false;
    if (primaryString.includes(comparedString.substr(0, seekLength))) return true;
    
    return checkForSharedSubstring(primaryString, comparedString.slice(1), seekLength);
}

File: C#/sharedSubstring.cs
/*
    Returns true on finding a shared substring of length seekLength, else returns false.
*/

namespace achorsnippets 
{
    public static bool CheckForSharedSubstring(string primaryString, string comparedString, int seekLength)
		{
			if (comparedString.Count() < seekLength)
			{
				return false;
			}
			else if (primaryString.Contains(comparedString.Substring(0, seekLength)))
			{
				return true;
			}

			return CheckForSharedSubstring(primaryString, comparedString.Substring(1), seekLength);
		}
}